icu4c/source/i18n/vtzone.cpp

```diff
@@ -1380,21 +1422,23 @@ VTimeZone::parse(UErrorCode& status) {
     int32_t finalRuleCount = 0;
 
     rules = new UVector(status);
+    if (rules == NULL) {
+        status = U_MEMORY_ALLOCATION_ERROR;
+    }
     if (U_FAILURE(status)) {
         goto cleanupParse;
     }
      // Set the deleter to remove TimeZoneRule vectors to avoid memory leaks due to unowned TimeZoneRules.
     rules->setDeleter(deleteTimeZoneRule);
     
     dates = new UVector(uprv_deleteUObject, uhash_compareUnicodeString, status);
-    if (U_FAILURE(status)) {
-        goto cleanupParse;
-    }
-    if (rules == NULL || dates == NULL) {
+    if (rules == NULL) {
```

Comment:
`dates`

Reply:
Oh good catch. Thanks!
(I'll force-push an update later with NULL->nullptr changes too.)

Reply:
Actually, for review purposes, I'll probably push more commits then squash at the end. (Though it will require more approvals though...)